src/components/ec-timer/ec-timer.vue

```diff
@@ -0,0 +1,131 @@
+<template>
+  <div
+    class="ec-timer"
+    data-test="ec-timer"
+  >
+    <svg
+      :width="diameter"
+      :height="diameter"
+      :viewBox="viewbox"
+      class="ec-timer__svg"
+    >
+      <circle
+        class="ec-timer__background"
+        cx="50%"
+        cy="50%"
+        :r="radius"
+        fill="transparent"
+        :stroke-width="strokeWidth"
+      />
+
+      <circle
+        class="ec-timer__remaining"
+        cx="50%"
+        cy="50%"
+        :r="radius"
+        fill="transparent"
+        :stroke-width="strokeWidth"
+        :stroke-dasharray="circumference"
+        :stroke-dashoffset="offset"
+      />
+
+      <text
+        x="50%"
+        y="50%"
+        class="ec-timer__text"
+        dominant-baseline="central"
+        text-anchor="middle"
+      >
+        {{ secondsLeft }}s
+      </text>
+    </svg>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'EcTimer',
+  props: {
+    seconds: {
+      type: Number,
+      required: true,
+      validator(value) {
+        return Number.isInteger(value) && value > 0;
+      },
+    },
+    isRunning: {
+      type: Boolean,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      radius: 24,
+      strokeWidth: 4,
+      secondsLeft: this.seconds,
+    };
+  },
+  computed: {
+    diameter() {
+      return (this.radius * 2) + this.strokeWidth;
+    },
+    viewbox() {
+      return `0 0 ${this.diameter} ${this.diameter}`;
+    },
+    circumference() {
+      return 2 * Math.PI * this.radius;
+    },
+    steps() {
+      return this.circumference / this.seconds;
+    },
+    offset() {
+      return this.circumference + this.steps * this.secondsLeft;
+    },
+  },
+  watch: {
+    isRunning: {
+      immediate: true,
+      handler(value) {
+        if (value) {
+          this.countdown();
+        } else {
+          this.secondsLeft = this.seconds;
```

Comment:
@stavros-tomas , if when the timer is counting down we set to `false` the `isRunning` prop, the `secondsLeft` variable is reset (as we expect), but the timeout still keeps working, so the event is emitted at the end anyway. In this scenario, we should remove the timeout also in order to avoid emitting the `time-expired` event.